File: app/src/main/java/net/daverix/ajvm/ALoadOperation.java
package net.daverix.ajvm;


import net.daverix.ajvm.io.ByteCodeReader;
import net.daverix.ajvm.jvm.ByteCodeOperation;
import net.daverix.ajvm.jvm.Frame;

import java.io.IOException;

class ALoadOperation implements ByteCodeOperation {
    @Override
    public void execute(ByteCodeReader reader, int indexOfBytecode, Frame currentFrame) throws IOException {
        int index = reader.readUnsignedByte();
        Object variable = currentFrame.getLocalVariable(index);
        currentFrame.push(variable);
    }
}

File: app/src/main/java/net/daverix/ajvm/operation/ALoadOperation.kt
package net.daverix.ajvm.operation


import net.daverix.ajvm.Frame
import net.daverix.ajvm.ByteCodeReader

import java.io.IOException

class ALoadOperation : ByteCodeOperation {
    @Throws(IOException::class)
    override fun execute(reader: ByteCodeReader, indexOfBytecode: Int, currentFrame: Frame) {
        val index = reader.readUnsignedByte()
        val variable = currentFrame.getLocalVariable(index)
        currentFrame.push(variable!!)
    }
}
